pygen/pygen_src/riscv_data_page_gen.py

```diff
@@ -0,0 +1,87 @@
+"""
+Copyright 2020 Google LLC
+Copyright 2020 PerfectVIPs Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+"""
+
+import logging
+import vsc
+import random
+from collections import defaultdict
+from pygen_src.riscv_instr_pkg import pkg_ins, data_pattern_t
+from pygen_src.riscv_instr_gen_config import cfg
+
+
+@vsc.randobj
+class riscv_data_page_gen:
+    def __init__(self):
+        self.data_page_str = []
+        self.mem_region_setting = defaultdict(list)
+
+    def gen_data(self, idx, pattern, num_of_bytes, data):
```

Comment:
`gen_data` seems to be a `staticmethod`. If so, please define it as a static method.

Reply:
Hi @hodjat91 
It is fixed now!